src/prefect/infrastructure/docker.py

```diff
@@ -39,6 +40,63 @@ class ImagePullPolicy(AutoEnum):
     NEVER = AutoEnum.auto()
 
 
+class DockerRegistry(Block):
+    """
+    Connects to a Docker registry.
+
+    Requires a Docker Engine to be connectable. Login information is persisted to disk
+    at the Docker default location.
+
+    Attributes:
+        username: The username to log into the registry with.
+        password: The password to log into the registry with.
+        registry_url: The URL to the registry. Generally, "http" or "https" can be
+            omitted.
+        reauth: If already logged into the registry, should login be performed again?
+            This setting defaults to `True` to support common token authentication
+            patterns such as ECR.
+    """
+
+    _block_type_name = "Docker Registry"
+    _block_schema_capabilities = ["login"]
```

Comment:
I think the capability should be a little more specific
```suggestion
    _block_schema_capabilities = ["docker-login"]
```

Reply:
Agreed.